velox/exec/tests/PlanNodeToStringTest.cpp

```diff
@@ -422,25 +425,47 @@ TEST_F(PlanNodeToStringTest, exchange) {
 }
 
 TEST_F(PlanNodeToStringTest, tableScan) {
-  RowTypePtr rowType{ROW(
-      {"discount", "quantity", "shipdate"}, {DOUBLE(), DOUBLE(), VARCHAR()})};
+  RowTypePtr rowType{
+      ROW({"discount", "quantity", "shipdate", "comment"},
+          {DOUBLE(), DOUBLE(), VARCHAR(), VARCHAR()})};
   auto filters = SubfieldFiltersBuilder()
                      .add("shipdate", between("1994-01-01", "1994-12-31"))
                      .add("discount", betweenDouble(0.05, 0.07))
                      .add("quantity", lessThanDouble(24.0))
                      .build();
-  auto tableHandle = exec::test::HiveConnectorTestBase::makeTableHandle(
-      std::move(filters), nullptr, "lineitem");
-  auto plan =
-      PlanBuilder()
-          .tableScan(
-              rowType,
-              tableHandle,
-              exec::test::HiveConnectorTestBase::allRegularColumns(rowType))
-          .planNode();
-
-  ASSERT_EQ("-> TableScan\n", plan->toString());
-  ASSERT_EQ(
-      "-> TableScan[Table: lineitem, Filters: [(discount, DoubleRange: [0.050000, 0.070000] no nulls), (quantity, DoubleRange: (-inf, 24.000000) no nulls), (shipdate, BytesRange: [1994-01-01, 1994-12-31] no nulls)]]\n",
-      plan->toString(true, false));
+  auto untyped = facebook::velox::duckdb::parseExpr(
+      "comment NOT LIKE '%special%request%'");
+  auto remainingFilter =
+      core::Expressions::inferTypes(untyped, rowType, nullptr);
+  {
+    auto tableHandle = exec::test::HiveConnectorTestBase::makeTableHandle(
+        std::move(filters), remainingFilter, "lineitem");
+    auto plan =
+        PlanBuilder()
+            .tableScan(
+                rowType,
+                tableHandle,
+                exec::test::HiveConnectorTestBase::allRegularColumns(rowType))
+            .planNode();
+
+    ASSERT_EQ("-> TableScan\n", plan->toString());
+    ASSERT_EQ(
+        "-> TableScan[Table: lineitem, Filters: [(discount, DoubleRange: [0.050000, 0.070000] no nulls), (quantity, DoubleRange: (-inf, 24.000000) no nulls), (shipdate, BytesRange: [1994-01-01, 1994-12-31] no nulls), (not(like(ROW[\"comment\"],\"%special%request%\")))]]\n",
+        plan->toString(true, false));
+  }
+  {
+    auto tableHandle = exec::test::HiveConnectorTestBase::makeTableHandle(
+        {}, remainingFilter, "lineitem");
+    auto plan =
+        PlanBuilder()
+            .tableScan(
+                rowType,
+                tableHandle,
+                exec::test::HiveConnectorTestBase::allRegularColumns(rowType))
+            .planNode();
+
+    ASSERT_EQ(
+        "-> TableScan[Table: lineitem, Filters: [(not(like(ROW[\"comment\"],\"%special%request%\")))]]\n",
```

Comment:
Let's use lowercase "table" and "filters". Also, it is often useful to separate range filters and remaining filter.